src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const res = [];
+
+  for (const i of actions) {
+    if (i.type === 'addProperties') {
+      Object.assign(stateClone, i.extraData);
+    }
+
+    if (i.type === 'removeProperties') {
+      for (const j of i.keysToRemove) {
+        delete stateClone[j];
+      }
+    }
+
+    if (i.type === 'clear') {
+      Object.keys(stateClone).forEach((key) => delete stateClone[key]);
+    }
+
+    res.push({ ...stateClone });
```

Comment:
You are creating a new copy of `stateClone` each time you push it to `res`. This is unnecessary as you already created a copy of `state` at the beginning. You can just push `stateClone` directly.